kano_video/ui/video.py

```diff
@@ -56,11 +57,14 @@ def __init__(self, e, playlist_name=None, permanent=False):
         self.add(button_grid)
 
         img = Gtk.Image()
-
+        
         if e['thumbnail']:
-            thumbnail = '{}/video_{}.jpg'.format(tmp_dir, time())
-            download_url(e['thumbnail'], thumbnail)
-            img.set_from_file(thumbnail)
+            if os.path.exists(e['thumbnail']):# use local thumbnail if one exists
+                img.set_from_file(e['thumbnail'])
+            else: #download the file if local does not exist
+                thumbnail = '{}/video_{}.jpg'.format(tmp_dir, time())
+                download_url(e['thumbnail'], thumbnail)
+                img.set_from_file(thumbnail)
```

Comment:
Those modifications are used to display local thumbnails. e['thumbnail'] can be a URL but it can also point to a local file in playlists/Library.json   I had a problem with viewing the thumbnails from the Library playlist so I added that bit of code to display local thumbnails if they exist, pointed by the thumbnail field in playlists/Library.json It is meant to work with https://github.com/KanoComputing/kano-video/pull/57